internal/controller/rbac/provider/roles/roles.go

```diff
@@ -123,6 +135,12 @@ func RenderClusterRoles(pr *v1.ProviderRevision, crds []extv1.CustomResourceDefi
 		Rules: withVerbs(rules, verbsEdit),
 	}
 
+	// The rule grants access to update the finalizers of resources that belong to the API group.
+	// If OwnerReferencesPermissionEnforcement is enabled, it allows after deleting the provider to
+	// clean up resources that they no longer own, so that the system can be kept in a consistent state.
+	// See https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
+	edit.Rules = append(edit.Rules, withVerbs(finalizerRules, verbsUpdate)...)
```

Comment:
> it allows after deleting the provider to clean up resources that they no longer own

Is this comment accurate? It sounds like you're saying that this somehow helps with deleting a `Provider.pkg`?

I might be wrong, but I _think_ what you're actually trying to do here is allow Crossplane access to update the finalizers of a managed resource. The only reason I can think of that Crossplane would need this permission would be to create connection secrets that would block the deletion of a managed resource. If that's the case, I think this rule should apply to the `system` role rather than `edit`. The `system` role is what the provider pod runs as, and it's the provider pod that would be creating managed resource connection secrets.

If I'm right, I would prefer the comment to be more explicit, for example:

> Provider pods may create Kubernetes secrets containing managed resource connection details. These secrets are controlled (in the owner reference sense) by the managed resource. Crossplane needs permission to set finalizers on managed resources in order to create secrets that block their deletion when the OwnerReferencesPermissionEnforcement admission controller is enabled.

Reply:
> The only reason I can think of that Crossplane would need this permission would be to create connection secrets that would block the deletion of a managed resource.

@el-mail I see you've removed this rule entirely. Did you determine it wasn't needed? Have you confirmed in your testing that managed resource connection secrets are being written successfully?

Reply:
@negz this is needed.
Without this permission, I've got this error at MR:
`connect failed: cannot track ProviderConfig usage: cannot apply
        ProviderConfigUsage: cannot create object:
        providerconfigusages.aws.crossplane.io
        "1220bf14-83ce-484b-a321-dc75495c6227" is forbidden: cannot set
        blockOwnerDeletion if an ownerReference refers to a resource you can't
        set finalizers on: , <nil>`

https://github.com/crossplane-contrib/provider-aws/blob/c9ed477f32e6e22b2c7ac7ed251b53ca927b5054/pkg/clients/aws.go#L420